File: src/main/java/org/mitallast/queue/security/SecurityModule.java
package org.mitallast.queue.security;

import com.google.inject.AbstractModule;
import org.mitallast.queue.common.codec.Codec;

public class SecurityModule extends AbstractModule {
    static {
        Codec.Companion.register(10, ECDHRequest.class, ECDHRequest.codec);
        Codec.Companion.register(11, ECDHResponse.class, ECDHResponse.codec);
        Codec.Companion.register(12, ECDHEncrypted.class, ECDHEncrypted.codec);
    }

    @Override
    protected void configure() {
        bind(SecurityService.class).asEagerSingleton();
    }
}

File: src/main/java/org/mitallast/queue/security/SecurityModule.kt
package org.mitallast.queue.security

import com.google.inject.AbstractModule
import org.mitallast.queue.common.codec.Codec

class SecurityModule : AbstractModule() {

    override fun configure() {
        bind(SecurityService::class.java).asEagerSingleton()
    }

    companion object {
        init {
            Codec.register(10, ECDHRequest::class.java, ECDHRequest.codec)
            Codec.register(11, ECDHResponse::class.java, ECDHResponse.codec)
            Codec.register(12, ECDHEncrypted::class.java, ECDHEncrypted.codec)
        }
    }
}
